test/Mono.Linker.Tests.Cases/Reflection/UnseenCallersAnnotation.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Reflection;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.Cases.Expectations.Metadata;
+
+namespace Mono.Linker.Tests.Cases.Reflection
```

Comment:
I would move this out into it's own top level directory and add a new test method w/ `[TestCaseSource]` in `TestSuites.cs`.    Two reasons

1) To date, the `Reflection` folder has been focused on tests for the detection & marking of reflection usage.  I feel like this is a separate behavior from that.
2) It will be easy to ignore all of these tests on Windows if they all fall under a single `[TestCaseSource]` method .  See https://github.com/mono/linker/pull/506/files#r270871566